satellite/mailservice/service.go

```diff
@@ -74,12 +77,28 @@ func New(log *zap.Logger, sender Sender, templatePath string) (*Service, error)
 	return service, nil
 }
 
+// Close closes and waits for any pending actions.
+func (service *Service) Close() error {
```

Comment:
`Close` isn't a good name for this method because it doesn't close anything, it only waits.

I usually expect that a `Close` method finalizes something and the instance gets partially/totally unutilizable.

Reply:
Yeah, `Close` is currently used as a thing to say "please ensure that this service has completely finished". Having different names like `WaitForFinish`, `Finish` and `Stop` ended up causing more confusion... the semantics for `Close` are more clearer - if something has `Close` you are expected to call it when you are done with it.

But, yeah the comment could be clearer on what it does.

Reply:
I don't see it in that way, I see that's a bad name for what it does, but anyway, it's my personal perception.

Reply:
I agree about the name not being descriptive enough. I.e. the consistency for using "Close" to finalize made working with peer code easier. Initially there were three or four different methods to show the difference, so I was never sure what was the correct method to call to tear down a service.